src/fsharp/utils/prim-lexing.fs

```diff
@@ -309,7 +309,7 @@ namespace Internal.Utilities.Text.Lexing
         let afterRefill (trans: uint16[][],sentinel,lexBuffer:LexBuffer<char>,scanUntilSentinel,endOfScan,state,eofPos) = 
             // end of file occurs if we couldn't extend the buffer 
             if lexBuffer.BufferScanLength = lexBuffer.BufferMaxScanLength then  
-                let snew = int trans.[state].[eofPos] // == EOF 
+                let snew = int trans[state].[eofPos] // == EOF 
```

Comment:
@dsyme Only one of the dots can be removed here, otherwise there's an error. Is it intended or a bug?

Reply:
Without the dot it is surprisingly parsed like this:
```
app
  app
    `int`
    app
      `trans`
      `[state]`
  `[eofPos]`
```
https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWTxhUDaiAXV4ArAQB1kkEAF8gA

<img width="996" alt="Screenshot 2022-04-08 at 21 49 27" src="https://user-images.githubusercontent.com/3923587/162503529-975378e6-1421-4131-9693-d7b9043e327c.png">

Looking at the warning, it seems expected 😞
Would it be feasible to allow nested indexing without parens? Otherwise it works not in a way one might really expect here.